### ISSUE TITLE
Better Type Inference in the IDE during Async/Await Error scenarios ? 

### ISSUE THREAD

**Issue by balajikris** [dotnet/roslyn#3951]
```C#
using System;
class Program
{
    static Task Test()
    {
        B b = new B();
        var result = await TestAsync();
        switch$$
    }

    static async Task<ConsoleColor> TestAsync()
    {
        throw new NotImplementedException();
    }
}
```

At $$, type `<TAB>result<ENTER>` (we're invoking switch snippet, typing in identifier name as `result` and the expanding snippet body and committing the session via hitting Enter).

**Expected**: the switch body is expanded with all the cases for ConsoleColor.
**Actual**: only default case is inserted. 
```C#
switch (result)
{
    default:
        break;
}
```

__Notes__:
It looks like this happens because the assignment to `result` in the previous line did not bind and the IDE was unable to infer the type of the identifier. Fixing the error by adding the `async` modifier to `Test` Method and then expanding the snippet works fine. Can we improve the experience here?

**Comment by Pilchie** [comment]
We already parse this as an await expression - can we have the rest of the machinery work in the fault tolerance case?

**Comment by jaredpar** [comment]
Why is this filed as a compiler issue?  Is the expectation that we set the type of `result` as if the `await` expression bound successfully?

**Comment by Pilchie** [comment]
Right - in  a non-async method, `SemanticModel.GetSymbolInfo` is not unwrapping the type of an `await` expression, even though it did parse it that way.  I can understand not getting results back if it's a place where the parse is different because it's not an async method, but it seems like if we're going to parse as an `await`, we should be able to bind that way too.

**Comment by gafter** [comment]
Closed in #10232.
